Rotational: default values are now guesses + fix tests

ven-k opened the pull request:
- Defaults in rotational lib are `guess`es
- Passes du0 and u0 for tests (in Rotational, Blocks)
- Fixes the linear analysis tests and docs (related to DC Motor example)

## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [x] All documentation related to code changes were updated
- [x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [x] Any new documentation only uses public API

ven-k commented on test/Mechanical/rotational.jl:
```diff
@@ -8,215 +8,219 @@ using OrdinaryDiffEq: ReturnCode.Success
+    @mtkbuild sys = TwoInertia()
 
-    prob = ODEProblem(sys, Pair[], (0, 10.0))
+    prob = ODEProblem(sys, [D(D(sys.inertia2.phi)) => 0], (0, 10.0))
```
@YingboMa this and in following ODE and DAE problems, now we have to pass an initial value to `sys₊inertia2₊phi_tt`. However, while adding this lib this was discussed and after some changes in tearing (in MTK), this was no longer required.
Why is it required again?

ven-k commented:
@ChrisRackauckas this is ready for review.

baggepinnen commented on src/Blocks/continuous.jl:
```diff
@@ -12,11 +12,15 @@ Initial value of integrator state ``x`` can be set with `x`
     @extend u, y = siso = SISO()
     @variables begin
-        x(t) = 0.0, [description = "State of Integrator"]
+        x(t), [description = "Unknown of Integrator", guess = 0.0]
```
This instance of "state" should not be translated to "unknown", state is indeed correct here

ven-k replied:
Resolved here: https://github.com/SciML/ModelingToolkitStandardLibrary.jl/pull/274/commits/eae592ba86e9de11fa575f005d3c13b526a80c99#diff-2c628e949fda2edfd7af3ee31fbf986535a6b1e959d92d03be9ca070a4b633aeR23

ChrisRackauckas commented on test/Blocks/test_analysis_points.jl:
```diff
@@ -165,7 +166,7 @@ function SystemModel(u = nothing; name = :model)
     if u !== nothing
         push!(eqs, connect(torque.tau, u.output))
-        return @named model = ODESystem(eqs, t;
+        return ODESystem(eqs, t;
```
then the name needs to be passed

ven-k replied:
Resolved here: https://github.com/SciML/ModelingToolkitStandardLibrary.jl/pull/274/commits/eae592ba86e9de11fa575f005d3c13b526a80c99#diff-3603e44df7347f59dcb8354b6aa1af55ec51d86974ae7a5521ae7214cba54c27R177-R178

ven-k commented on src/Mechanical/Rotational/components.jl:
```diff
@@ -167,7 +167,7 @@ Linear 1D rotational spring and damper
         d, [description = "Damping constant"]
         c, [description = "Spring constant"]
-        phi_rel0 = 0.0, [description = "Unstretched spring angle"]
+        phi_rel0, [description = "Unstretched spring angle", guess = 0.0]
```
This is an oversight; will revert.

ven-k replied:
Resolved here: https://github.com/SciML/ModelingToolkitStandardLibrary.jl/pull/274/commits/eae592ba86e9de11fa575f005d3c13b526a80c99#diff-36a0d1920684771c0166099143e51f1cf58f1faae7d847cb151c0a0826c4348bR170

ven-k commented:
Downgrade CI doesn't seem to be happy yet; but I will recheck up to which versions I can bump them to (as a follow up). I don't want to increase the compats to latest available versions; but keep them just above the minimum required ones.

ven-k commented:
The failing Downgrade CI test is fixed by:
- https://github.com/SciML/ModelingToolkitStandardLibrary.jl/pull/276

baggepinnen commented:
Considering how much of the user-facing documentation and tests had to be changed in this PR, I'd say this is a massively breaking change that requires a new major version?

baggepinnen commented on src/Blocks/continuous.jl:
```diff
@@ -12,11 +12,15 @@ Initial value of integrator state ``x`` can be set with `x`
     @extend u, y = siso = SISO()
     @variables begin
-        x(t) = 0.0, [description = "State of Integrator"]
+        x(t), [description = "State of Integrator", guess = 0.0]
```
There is no guess needed here, this should be a default value.

ven-k replied:
Resolved with: https://github.com/SciML/ModelingToolkitStandardLibrary.jl/pull/274/commits/3a7ccfbd43de07b1f1de3f81dc1b86e173146eac

baggepinnen commented on src/Blocks/continuous.jl:
```diff
@@ -57,7 +66,7 @@ A smaller `T` leads to a more ideal approximation of the derivative.
     @extend u, y = siso = SISO()
     @variables begin
-        x(t) = 0.0, [description = "Derivative-filter state"]
+        x(t), [description = "Derivative-filter state", guess = 0.0]
```
Same here, there is no guessing involved, a default is much better.

ven-k replied:
Resolved with: https://github.com/SciML/ModelingToolkitStandardLibrary.jl/pull/274/commits/3a7ccfbd43de07b1f1de3f81dc1b86e173146eac

ven-k commented:
> I'd say this is a massively breaking change that requires a new major version?

This sounds good. Even the Units PR will be unblocked with this PR. So I'd say we can have that in too.
(And probably guesses in other sub-libraries as well - with dedicated PRs for each libraries)

> Was there a particular reasons these two blocks and only these were changed?

The idea is - incrementally change all the sub-libraries; tend to the ones with broken tests first.

The analysis-point tests were failing as the rotational components now have guesses instead of defaults - fixed with `op` - while there, I changed these two as they were involved.

I've reverted the changes to `Integrator` and `Derivative` blocks.

baggepinnen commented:
This was released in a minor version but it should have been a major version bump 😔

ChrisRackauckas commented:
It probably should have.
